Include cherry-picked patches from upstream to fix invalid event pointers.

--- a/window-copy.c
+++ b/window-copy.c
@@ -171,7 +171,8 @@
 	data->searchstr = NULL;
 
 	wp->flags |= PANE_FREEZE;
-	bufferevent_disable(wp->event, EV_READ|EV_WRITE);
+	if (wp->fd != -1)
+		bufferevent_disable(wp->event, EV_READ|EV_WRITE);
 
 	data->jumptype = WINDOW_COPY_OFF;
 	data->jumpchar = '\0';
@@ -234,7 +235,8 @@
 	struct window_copy_mode_data	*data = wp->modedata;
 
 	wp->flags &= ~PANE_FREEZE;
-	bufferevent_enable(wp->event, EV_READ|EV_WRITE);
+	if (wp->fd != -1)
+		bufferevent_enable(wp->event, EV_READ|EV_WRITE);
 
 	if (data->searchstr != NULL)
 		xfree(data->searchstr);
